pkg/proxy/policy/policy.go

```diff
@@ -0,0 +1,61 @@
+// Package policy defines the network policy that the proxy can choose to enforce.
+package policy
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/elazarl/goproxy"
+)
+
+type MatchingType int
+
+const (
+	PathPrefix MatchingType = iota
+	FullPath
+)
+
+type Policy struct {
+	Rules []Rule
+}
+
+func (p Policy) EnforcePolicy(req *http.Request, ctx *goproxy.ProxyCtx) (*http.Request, *http.Response) {
+	url := req.URL
+	for _, rule := range p.Rules {
+		if rule.IsCompliant(req) {
```

Comment:
hmm I actually think it's tricky to understand what to expect with these names. A "policy" with many "rule"s which evaluate "compliance" makes me expect that a request must pass all rules to satisfy the policy. Is there another way we could assign names here to make this less surprising?

Maybe just renaming "Policy.Rules" to "Policy.Allowlist"? Or maybe "Policy.AllowAny"? Or "Policy.AnyOf"